### ISSUE TITLE
Model checkpointing is done by validation loss, not validation accuracy

### ISSUE THREAD

**Issue by matt-gardner** [allenai/deep_qa#199]
This means that the "best epoch" as defined by our code may not have actually been saved by Keras.  We need to be sure that the model checkpointing uses the same metric as early stopping.
